[Workers] Module Workers docs (#9265)
cloudflare/cloudflare-docs: content/workers/learning/migrate-to-module-workers.md

```js
---
filename: todos.js
---
async function getTodos(env) {
  // NOTE: Relies on the TODO KV binding which has been provided inside of 
  // the env parameter of the `getTodos` function
  let value = await env.TODO.get("to-do:123");
  return new Response(value);
}

export { getTodos }
```

## Access `event` or `context` data

Workers often need access to data not in the `request` object. For example, sometimes Workers use [`waitUntil`](/workers/runtime-apis/fetch-event/#waituntil) to delay execution. Workers using ES modules format can access `waitUntil` via the `context` parameter. Refer to [ES modules parameters](/workers/runtime-apis/fetch-event/#parameters) for  more information.

This example code:

```js
async function triggerEvent(event) {
  // Fetch some data
  console.log('cron processed', event.scheduledTime);
}

// Initialize Worker
addEventListener('scheduled', event => {
  event.waitUntil(triggerEvent(event));
});
```

Then becomes:

```js
async function triggerEvent(event) {
  // Fetch some data
  console.log('cron processed', event.scheduledTime);
}

export default {
  async scheduled(event, env, ctx) {
    ctx.waitUntil(triggerEvent(event));
  },
};

> 
> export default worker;

Reviewer: ```suggestion
```

---
pcx_content_type: concept

> title: Migrate to module Workers

Reviewer 2: ```suggestion
title: Migrate from the Service Worker to ES Module format
```

weight: 9
---

> # Migrate to module Workers

Reviewer 2: ```suggestion
# Migrate from the Service Worker to ES Module format
```

> This guide will show you how to migrate your Workers from the [Service Worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) syntax to the new [Module Worker](https://blog.cloudflare.com/workers-javascript-modules/) format.

Reviewer 2: ```suggestion
This guide will show you how to migrate your Workers from the [Service Worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) foramt to the [ES Modules](https://blog.cloudflare.com/workers-javascript-modules/) format. 
```

## Advantages of migrating

> There are several reasons you might want to migrate your Workers to the module syntax:

Reviewer 2: ```suggestion
There are several reasons you might want to migrate your Workers to the ES Modules foramt:
```

> 1.  Developer Platform products, such as [Durable Objects](/workers/learning/using-durable-objects/), and other Workers features require the module syntax.

Reviewer 2: ```suggestion
1.  Many products within Cloudflare's Developer Platform, such as [Durable Objects](/workers/learning/using-durable-objects/), and other features of Cloudflare Workers, require the ES Modules format.
```

> 2.  Module Workers do not rely on any global bindings, which means the Workers runtime does not need to set up fresh execution contexts, making Module Workers safer and faster to run.

Reviewer 2: General note — I'm not sure we want to refer to these as "Module Workers", comes across more like this is a Capital Letter feature or product in and of itself — seems most specific to say "ES module format" — thoughts?

  },
};
```

The following code represents a `getTodos` function that calls the `get` function on the `TODO` KV binding.

```js
---
filename: todos.js
---
async function getTodos(env) {
  // NOTE: Relies on the TODO KV binding which has been provided inside of 
  // the env parameter of the `getTodos` function
  let value = await env.TODO.get("to-do:123");
  return new Response(value);
}

export { getTodos }
```

## Accessing event or context data

Workers often need access to data not in the `request` object. For example, sometimes Workers use [`waitUntil`](/workers/runtime-apis/fetch-event/#waituntil) to delay execution. Module workers can access `waitUntil` via the `context` parameter. Refer to [Module Worker parameters](/workers/runtime-apis/fetch-event/#parameters) for  more information.

This example code:

```js
async function triggerEvent(event) {
  // Fetch some data
  console.log('cron processed', event.scheduledTime);
}

// Initialize Worker
addEventListener('scheduled', event => {
  event.waitUntil(triggerEvent(event));
});
```

Then becomes:

```js
async function triggerEvent(event) {
  // Fetch some data
  console.log('cron processed', event.scheduledTime);
}

> const worker = {

Reviewer 2: This line could just be the default export? Simpler?

Author: ```suggestion
export default {
```
@irvinebroque does this work?

  async scheduled(event, env, ctx) {
    ctx.waitUntil(triggerEvent(event));
  },
};

export default worker;
```

## Configure your wrangler.toml

To add support for module Workers to an existing Workers project, update your `wrangler.toml` file:

```toml
name = "my-worker"
main = "src/index.js"

# Uncomment if you have a build script.
# [build]
# command = "npm run build"

> ```

Reviewer 2: Can remove this section
